File: core/src/main/java/com/iwdael/dbroom/core/Persistence.java
package com.iwdael.dbroom.core;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import org.jetbrains.annotations.NotNull;

/**
 * @author  : iwdael
 * @mail    : iwdael@outlook.com
 * @project : https://github.com/iwdael/dbroom
 */
@Entity(
        tableName = "persistence"
)
public class Persistence {
    @PrimaryKey
    @ColumnInfo(name = "persistence_name")
    @NotNull
    public String name;

    @ColumnInfo(name = "persistence_value")
    public String value;

    public Persistence(@NonNull String name, String value) {
        this.name = name;
        this.value = value;
    }
}

File: core/src/main/java/com/iwdael/dbroom/core/Persistence.kt
package com.iwdael.dbroom.core

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * @author  : iwdael
 * @mail    : iwdael@outlook.com
 * @project : https://github.com/iwdael/dbroom
 */
@Entity("persistence")
class Persistence {
    @PrimaryKey
    @ColumnInfo(name = "persistence_name")
    var name: String

    @ColumnInfo(name = "persistence_value")
    var value: String? = null

    constructor(name: String, value: String?) {
        this.name = name
        this.value = value
    }

    constructor() {
        this.name = ""
    }
}
